src/features/chat/Chat.tsx

```diff
@@ -0,0 +1,263 @@
+    }
+
+    return ''
+  }
+
+  const onKeypress = (e: KeyboardEvent<HTMLTextAreaElement>) => {
+    if (['Enter', 'NumpadEnter'].includes(e.code) && !e.shiftKey) {
+      e.preventDefault()
+      e.stopPropagation()
+
+      onSendMsg()
+    }
+  }
+
+  useEffect(() => {
+    scroll2End()
+  }, [chats, activeChat, activeChat.messages.length])
+
+  return (
+    <div
+      className='page-container py-5 flex flex-col h-full'
+      onClick={onCloseChat}
+    >
+      <div
+        className={cn(styles.chat, 'paper flex flex-grow')}
+        onClick={ev => ev.stopPropagation()}
+      >
+        <div className={styles.colLeft}>
+          <div className={styles.title}>
+            <h1>Chat</h1>
+            <i onClick={onChatEdit}>
+              <img src={editIcon} />
+            </i>
+          </div>
+          <div className={styles.chatsList}>
+            {chats.map((chat, i) => (
+              <ChatItem
+                key={i}
+                {...chat}
+                onClick={onSelectChat}
+                isActive={activeChatId === chat.id}
+              />
+            ))}
+          </div>
+        </div>
+
+        <div className={styles.chatRight}>
+          <div className={styles.chatHeader}>
+            <div className={cn(styles.userAvatar, styles.userAvatarSm)}>
+              {activeChat && (
+                <i
+                  style={{ backgroundImage: `url(${chatAvatar(activeChat)})` }}
+                ></i>
+              )}
+              {/* insert user avatar */}
+            </div>
+
+            <div className={styles.chatTitle}>
+              {activeChat ? activeChat.title : ''}
+            </div>
+
+            <div className={styles.chatMenu} onClick={onActiveChatMenu}>
+              <img src={chatMenuIcon} />
+            </div>
+
+            <div className={styles.chatClose} onClick={onCloseChat}>
+              <img src={chatCloseIcon} />
+            </div>
+          </div>
+
+          <div className={styles.chatMessages}>
+            {activeChat &&
+              activeChat.messages.map((msg, i) => {
+                const props = {
+                  ...msg,
+                  showAvatar:
+                    !i || activeChat.messages[i - 1].sender !== msg.sender,
+                }
+                return (
+                  <ChatMessage
+                    key={i}
+                    {...props}
+                    onSaveAttachment={saveAttachment}
+                  />
+                )
+              })}
+            <div ref={endRef}></div>
+          </div>
+          <div className={styles.chatFooter}>
+            <div className={cn(styles.userAvatar, styles.userAvatarSm2)}>
+              <i style={{ backgroundImage: `url(${curUser.avatar})` }}></i>
+              {/* insert current user avatar */}
+            </div>
+
+            <div className={styles.chatNewMsg}>
+              <label className={newMsgPlaceholder ? '' : styles.hidden}>
```

Comment:
Consider using `hidden` html property:
```
<label hidden={!newMsgPlaceholder}>
```

It's better practice to use conditional rendering, I mean `{variable && <label ...>}` so React won't need to render redundant elements